crates/uv-resolver/src/requires_python.rs

```diff
@@ -169,6 +170,57 @@ impl RequiresPython {
     pub fn bound(&self) -> &Bound<Version> {
         &self.bound
     }
+
+    /// Returns this `Requires-Python` specifier as an equivalent marker
+    /// expression utilizing the `python_version` marker field.
+    ///
+    /// This is useful for comparing a `Requires-Python` specifier with
+    /// arbitrary marker expressions. For example, one can ask whether the
+    /// returned marker expression is disjoint with another marker expression.
+    /// If it is, then one can conclude that the `Requires-Python` specifier
+    /// excludes the dependency with that other marker expression.
+    ///
+    /// If this `Requires-Python` specifier has no constraints, then this
+    /// returns a marker tree that evaluates to `true` for all possible marker
+    /// environments.
+    pub fn to_marker_tree(&self) -> MarkerTree {
+        let (op, version) = match self.bound {
+            // If we see this anywhere, then it implies the marker
+            // tree we would generate would always evaluate to
+            // `true` because every possible Python version would
+            // satisfy it.
+            Bound::Unbounded => return MarkerTree::And(vec![]),
+            Bound::Excluded(ref version) => (Operator::GreaterThan, version.clone()),
+            Bound::Included(ref version) => (Operator::GreaterThanEqual, version.clone()),
+        };
+        // For the `python_version` marker expression, it specifically only
+        // supports truncate major/minor versions of Python. This means that
+        // a `Requires-Python: 3.10.1` is satisfied by `python_version ==
+        // '3.10'`. So for disjointness checking, we need to ensure that the
+        // marker expression we generate for `Requires-Python` doesn't try to
+        // be overly selective about the patch version. We do this by keeping
+        // this part of our marker limited to the major and minor version
+        // components only.
+        let version_major_minor_only = Version::new(version.release().iter().take(2));
+        let expr_python_version = MarkerExpression::Version {
+            key: MarkerValueVersion::PythonVersion,
+            // OK because a version specifier is only invalid
+            // when the version is local (which is impossible
+            // here) or if the operator is ~= (which is also
+            // impossible here).
+            specifier: VersionSpecifier::from_version(op, version_major_minor_only).unwrap(),
+        };
+        let expr_python_full_version = MarkerExpression::Version {
+            key: MarkerValueVersion::PythonFullVersion,
+            // For `python_full_version`, we can use the entire
+            // version as-is.
+            specifier: VersionSpecifier::from_version(op, version).unwrap(),
```

Comment:
Why can this one unwrap?

Reply:
Same reason as the one above. I duplicated the comment to make this clearer.

Although I suppose the constructor of `RequiresPython` permits any `Version` which means it could technically have a local segment. Likely a logic error, but I added `without_local()` to strip it from the version anyway.